pkg/models/registries/image.go

```diff
@@ -30,9 +31,24 @@ func (i *Image) Reference() string {
 	return i.Tag
 }
 
+type DockerURL struct {
+	*url.URL
+}
+
+func (u *DockerURL) StringWithoutScheme() string {
+	u.Scheme = ""
+	s := u.String()
+	return strings.Trim(s, "//")
+}
+
+func ParseDockerURL(rawurl string) (*DockerURL, error) {
+	url, _ := url.Parse(rawurl)
+	return &DockerURL{URL: url}, nil
```

Comment:
so, no error will ever return, why the function signature has error as return value? It's not good.